src/App.js

```diff
@@ -107,9 +107,16 @@ class App extends React.Component {
 
     // New DAO: need to reinit the wrapper
     if (locator.dao && (!prevLocator || locator.dao !== prevLocator.dao)) {
+      this.cancelWrapperSubscriptions()
```

Comment:
What do you think about `updateDao()` doing this automatically?

Reply:
Yes we can do that too, I moved it here to group the `cancelWrapperSubscriptions()` calls but it’s not a strong reason.

Reply:
I moved everything this logic to `updateDao()`, it makes things simpler!